client/layout/index.jsx

```diff
@@ -164,7 +164,8 @@ class Layout extends Component {
 				config.isEnabled( 'woocommerce/onboarding-oauth' ) &&
 				isWooOAuth2Client( this.props.oauth2Client ) &&
 				this.props.wccomFrom,
-			'is-nav-unification': isAutomatticTeamMember( this.props.teams ),
+			'is-nav-unification':
+				isAutomatticTeamMember( this.props.teams ) && config.isEnabled( 'nav-unification' ),
```

Comment:
I've noticed there are [other locations](https://github.com/Automattic/wp-calypso/search?q=%22nav-unification%22) where we check for the `nav-unification` feature flag, but we don't account for `isAutomatticTeamMember` there. 

At the same time, this PR enables the flag for all non-production environments. 

Is that the desired behavior? Also, see the next question.

Reply:
Yes this the desired behavior, as currently decided `nav-unification` should be enabled for all a12s. More about why we still use the flag here https://github.com/Automattic/wp-calypso/issues/49123#issuecomment-765353474